develop/blockchains/custom-blockchains/benchmarking.md

```diff
@@ -1,4 +1,183 @@
+    ```rust title="benchmarks.rs"
+    --8<-- 'code/develop/blockchains/custom-blockchains/benchmarking/frame-benchmark-macro.rs'
+    ```
+    For example, to register a pallet named `pallet_parachain_template` for benchmark testing, add it as follows: 
+    ```rust title="benchmarks.rs" hl_lines="3"
+    --8<-- 'code/develop/blockchains/custom-blockchains/benchmarking/frame-benchmark-macro.rs::3'
+    );
+    ```
+
+    ???warning "Updating `define_benchmarks!` macro is required"
+        If the pallet isn't included in the `define_benchmarks!` macro, the CLI cannot access and benchmark it later.
+
+2. Navigate to the runtime's `lib.rs` file and add the import for `benchmarks.rs` as follows: 
+
+    ```rust title="lib.rs"
+    #[cfg(feature = "runtime-benchmarks")]
+    mod benchmarks;
+    ```
+
+    The `runtime-benchmarks` feature gate ensures benchmark tests are isolated from production runtime code.
+
+### Run Benchmarks
+
+You can now compile your runtime with the `runtime-benchmarks` feature flag. This feature flag is crucial as the benchmarking tool will look for this feature being enabled to know when it should run benchmark tests. Follow these steps to compile the runtime with benchmarking enabled:
+
+1. Run `build` with the feature flag included
+
+    ```bash
+    cargo build --features runtime-benchmarks --release
+    ```
+
+2. Once compiled, run the benchmarking tool to measure extrinsic weights
+
+    ```sh
+    ./target/release/<node-binary-name> benchmark pallet \
+    --runtime <path-to-wasm-runtime> \
+    --pallet <name-of-the-pallet> \
+    --extrinsic '*' \
+    --steps 20 \
+    --repeat 10 \
+    --output weights.rs
+    ```
+
+    ??? info "Flag definitions"
+        - `--runtime` - the path to your runtime's Wasm
+        - `--pallet` - the name of the pallet you wish to benchmark. This pallet must be configured in your runtime and defined in `define_benchmarks`
+        - `--extrinsic` - which extrinsic to test. Using `'*'` implies all extrinsics will be benchmarked
+        - `--output` - where the output of the auto-generated weights will reside
+
+The generated `weights.rs` file contains weight annotations for your extrinsics, ready to be added to your pallet. The output should be similar to the following. Some output is omitted for brevity: 
+
+--8<-- 'code/develop/blockchains/custom-blockchains/benchmarking/benchmark-output.html'
+
+#### Add Benchmark Weights to Pallet
+
+Once the `weights.rs` is generated, you may add the generated weights to your pallet. It is common that `weights.rs` become part of your pallet's root in `src/`:
+
+```rust
+--8<-- 'code/develop/blockchains/custom-blockchains/benchmarking/weight-config.rs'
+```
+
+After which, you may add this to the `#[pallet::weight]` annotation in the extrinsic via the `Config`: 
+
+```rust hl_lines="2"
+--8<-- 'code/develop/blockchains/custom-blockchains/benchmarking/dispatchable-pallet-weight.rs'
+```
+
+## What's Next
```

Comment:
```suggestion
## Where to Go Next
```

This is the name we use on other pages for this type of section, so just to keep it consistent